src/1.4.x/lib/bindings.sc

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env amm-2.11
+
+import $file.helpers
+import scala.annotation.tailrec
+
+import akka.actor.{ ActorSystem, ActorRefFactory, Scheduler }
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.{Source, Sink}
+import akka.util.Timeout
+import com.codahale.metrics.MetricRegistry
+import mesosphere.marathon.Protos.StorageVersion
+import mesosphere.marathon.core.instance.Instance
+import mesosphere.marathon.metrics.Metrics
+import mesosphere.marathon.state.PathId
+import mesosphere.marathon.storage._
+import mesosphere.marathon.storage.migration.{StorageVersions, Migration}
+import mesosphere.marathon.storage.repository._
+import org.apache.curator.framework.CuratorFramework
+import org.apache.curator.framework.recipes.leader.LeaderLatch
+import org.rogach.scallop.ScallopOption
+import scala.collection.immutable.Seq
+import scala.concurrent.duration._
+import scala.concurrent.{Future, Await}
+
+case class StorageToolModule(
+  appRepository: AppRepository,
+  podRepository: PodRepository,
+  instanceRepository: InstanceRepository,
+  deploymentRepository: DeploymentRepository,
+  taskFailureRepository: TaskFailureRepository,
+  groupRepository: GroupRepository,
+  frameworkIdRepository: FrameworkIdRepository,
+  migration: Migration
+)
+
+object MarathonStorage {
+  def argsFromEnv: List[String] = {
+    "(?<!\\\\)( +)".r.split(sys.env.getOrElse("MARATHON_ARGS", "").trim).toList.filterNot(_ == "")
+  }
+}
+class MarathonStorage(args: List[String] = MarathonStorage.argsFromEnv) {
+  import helpers.Helpers._
+  implicit val actorSystem = ActorSystem()
+  implicit val actorMaterializer = ActorMaterializer()
+  implicit val scheduler: Scheduler = actorSystem.scheduler
+  implicit val timeout = Timeout(5.seconds)
+  implicit val executionContext = scala.concurrent.ExecutionContext.global
+
+  private class ScallopStub[A](name: String, value: Option[A]) extends ScallopOption[A](name) {
```

Comment:
What is `ScallopStub` doing?

Reply:
We use it to disallow the configuration of storeCache and versionCacheEnabled, both of which we set to false.